app/src/main/java/org/wikipedia/analytics/eventplatform/ImageRecommendationsEvent.kt

```diff
@@ -0,0 +1,20 @@
+package org.wikipedia.analytics.eventplatform
+
+class ImageRecommendationsEvent(private val lang: String, private val page_title: String, private val image_title: String,
+                                private val suggestion_source: String, private val response_type: Int, private val reasons: List<Int>,
+                                private val info_clicked: Boolean, private val details_clicked: Boolean, scrolled: Boolean,
+                                private val time_until_click_ms: Long, private val time_until_submit_ms: Long, private val user_text: String?,
+                                private val teacher_mode_enabled: Boolean) : Event(SCHEMA_NAME, STREAM_NAME) {
```

Comment:
Just curious, are these variables are common variables that will be used in multiple log events? If not, would it be possible to optimize a bit?

Reply:
The problem is that the class gets automatically serialized to send to MEP... so they all need to be fields on the class and in that order..

Reply:
Got it.